stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -0,0 +1,1056 @@
+                    &block_hash,
+                    block
+                        .get_coinbase_tx()
+                        .ok_or(ChainstateError::InvalidStacksBlock(
+                            "No coinbase transaction in tenure changing block".into(),
+                        ))?,
+                    chain_tip_consensus_hash,
+                    next_block_height,
+                    block_fees,
+                    0,
+                    total_burnt,
+                    burnchain_commit_burn,
+                    burnchain_sortition_burn,
+                    total_coinbase,
+                )
+                .expect("FATAL: parsed and processed a block without a coinbase"),
+            )
+        } else {
+            None
+        };
+
+        let matured_rewards_info = miner_payouts_opt
+            .as_ref()
+            .map(|(_, _, _, info)| info.clone());
+
+        let new_tip = Self::advance_tip(
+            &mut chainstate_tx.tx,
+            &parent_chain_tip.anchored_header,
+            &parent_chain_tip.consensus_hash,
+            &block.header,
+            chain_tip_consensus_hash,
+            chain_tip_burn_header_hash,
+            chain_tip_burn_header_height,
+            chain_tip_burn_header_timestamp,
+            scheduled_miner_reward.as_ref(),
+            user_burns,
+            miner_payouts_opt,
+            &block_execution_cost,
+            &total_tenure_cost,
+            block_size,
+            applied_epoch_transition,
+            burn_stack_stx_ops,
+            burn_transfer_stx_ops,
+            burn_delegate_stx_ops,
+            affirmation_weight,
+        )
+        .expect("FATAL: failed to advance chain tip");
+
+        chainstate_tx.log_transactions_processed(&new_tip.index_block_hash(), &tx_receipts);
+
+        monitoring::set_last_block_transaction_count(block.txs.len() as u64);
+        monitoring::set_last_execution_cost_observed(&block_execution_cost, &block_limit);
+
+        // get previous burn block stats
+        let (parent_burn_block_hash, parent_burn_block_height, parent_burn_block_timestamp) =
+            if block.is_first_mined() {
+                (BurnchainHeaderHash([0; 32]), 0, 0)
+            } else {
+                match SortitionDB::get_block_snapshot_consensus(burn_dbconn, &parent_ch)? {
+                    Some(sn) => (
+                        sn.burn_header_hash,
+                        sn.block_height as u32,
+                        sn.burn_header_timestamp,
+                    ),
+                    None => {
+                        // shouldn't happen
+                        warn!(
+                            "CORRUPTION: block {}/{} does not correspond to a burn block",
+                            &parent_ch, &parent_block_hash
+                        );
+                        (BurnchainHeaderHash([0; 32]), 0, 0)
+                    }
+                }
+            };
+
+        let epoch_receipt = StacksEpochReceipt {
```

Comment:
Now that we use the word "epoch" to refer to a set of consensus rules, can we call this something else? e.g. "StacksBlockReceipt?"

Reply:
Sure -- should I do this rename in this PR? (I can make sure it's in a single commit of its own)

Reply:
It doesn't have to happen here